substrate/client/network/src/litep2p/shim/bitswap.rs

```diff
@@ -101,4 +107,25 @@ impl<Block: BlockT> BitswapServer<Block> {
 			}
 		}
 	}
+
+	/// Takes extracted values instead of the CID directly to avoid version conflicts
+	fn is_cid_supported(&self, version_num: u64, size: usize, cid_str: String) -> bool {
```

Comment:
We decided to import Cid from litep2p and remove `cid` dependency from polkadot-sdk for now
```suggestion
	pub fn is_cid_supported(cid: &Cid) -> bool {
```

And please move this function to the `substrate/client/network/src/bitswap/mod.rs`.
Because there is another filtering: 

```
            if cid.version() != cid::Version::V1 ||
				cid.hash().code() != u64::from(cid::multihash::Code::Blake2b256) ||
				cid.hash().size() != 32
			{
				debug!(target: LOG_TARGET, "Ignoring unsupported CID {}: {}", peer, cid);
				continue
			}
```

and change it also to:
```
			if !is_cid_supported(&cid) {
				debug!(target: LOG_TARGET, "Ignoring unsupported CID {}: {}", peer, cid);
				continue
			}
```